include/swift/Option/FrontendOptions.td

```diff
@@ -535,6 +535,11 @@ def build_module_from_parseable_interface :
   Alias<compile_module_from_interface>,
   ModeOpt;
 
+def preserve_types_as_written_in_module_interface :
+  Flag<["-"], "preserve-types-as-written-in-module-interface">,
```

Comment:
Can you add this to the XCTest overlay build, too, since we know we need it there?